arbeitszeit_flask/filters.py

```diff
@@ -69,8 +71,9 @@ def icon_filter(
         # Treat empty icon_name as intentionally set null value
         if icon_name.strip() == "":
             return Markup("")
-        file_path = os.path.join(ICON_PATH, f"{icon_name}.html")
-        svg_content = reader(file_path)
+
+        file_name = f"{icon_name}.html"
```

Comment:
I also thought about this but then decided to not use a stripped icon name string for the rest of the function because the initial conditional check cares about empty strings that are manually and [intentionally used as null values](https://github.com/ida-arbeitszeit/arbeitszeitapp/blob/master/arbeitszeit_web/www/presenters/show_prd_account_details_presenter.py#L118) and which "eventually" have 1 or more whitespaces set by accident. But this is maybe YAGNI and to implicit on my part. Things should be explicit or fail (wrong icon file name by accident) in development would be maybe the better approach here. Therefore I suggest to instead remove `.strip()` in the initial conditional check.

What do you think?

Reply:
> In which case would you want to provide the empty string?

I added the empty string conditional check because existing presenter logic used an empty string to avoid adding HTML classes. This change maintains the historical decisions made for that presenter. For more context, please go to the concerning source code linked in [my initial comment above](https://github.com/ida-arbeitszeit/arbeitszeitapp/blob/master/arbeitszeit_web/www/presenters/show_prd_account_details_presenter.py#L118).

In PR #1039, I adjusted the presenter to preserve the original intent. But instead of returning FontAwesome-specific classname strings like "fas fas-industry", "fas fas-user", or an empty string, it now returns "industry", "user", or an empty string for setting an icon in a specific macro. This macro icon approach was used in, if I recall correctly, 3 instances, while setting icons directly in dozens of template files was the default approach.

Personally, I would structure the view-model creation in a presenter in a way that no data about any to use icons is provided via a view-model to an HTML view function on the Flask framework layer from the presenter layer below. Instead a (Flask) HTML view function, aimed for human interaction, should be responsible to know what icons to render for stylistic purposes by using the data available in a generically structured view-model.

From a Clean Architecture perspective, I believe, the presenter should not handle any icon-related logic (inner-logic doesn't know anything about outer-logic). It should produce an generic view model that can be used in different view contexts, including those that don't involve styled interfaces for humans. An example would be a Data-API for machine-to-machine communication. These contexts don't require any stylistic decisions meant for human users.

However, I left this part out of PR #1039 because the PR was already huge and I didn't want to increase its scope any further.

Same here, I'm OK with leaving this PR as is or, as a minor change, remove `.strip()`. My main goal is to merge this PR to hopefully fix the icon issue we have in production. I suggest rolling any presenter icon discussion into another issue/PR if there is a need for this.

What do you think, sounds good?

Reply:
> So do I understand correctly that you would want to pass the empty string into the icon macro if you don't want to render an icon?

Yes, the idea of a template filter is that you pass a string and this string "can" get transformed into another string. In case of `{{ ""|icon }}` the empty string stays an empty string, hence, nothing happens.

Reply:
> Why would I want to do this?

You are right, it would make no sense, if one would type this out in a template, as source code. The use case for this is a different one.

My example was about using the filter with a string derived from a template context variable. As per your previous question ("pass the empty string into the icon `macro`") I assumed we talk about the same thing here, also because this topic was part of the previous conversation. Thats why I shorthanded the variable resolution in my icon filter example to `{{ ""|icon }}`.

I'm happy to explain the empty string aspect of the `icon` filter with more source code links to help paint a better picture here:

- The icon filter behaves like this because of a historical path dependency [in a presenter](https://github.com/ida-arbeitszeit/arbeitszeitapp/blob/16cedd496a479ddcecd9b4e4194394d58806f8a6/arbeitszeit_web/www/presenters/show_prd_account_details_presenter.py#L90-L103). The view model icon field this presenter generates is either a strong of "user", "industry" or empty string ("").
- The icon field is then used in [this template macro](https://github.com/ida-arbeitszeit/arbeitszeitapp/blob/16cedd496a479ddcecd9b4e4194394d58806f8a6/arbeitszeit_flask/templates/macros/transactions.html#L28-L53) in the icon template filter. The macro itself is used only once in another template.
- The presenter, and the icon filter in the template macro are therefore using the empty string as an explicit null value, my filter implementation follows this historical pre-defined logic by rendering, as before, an empty string if the input is an empty string.

As a side note, I you look at the template files I'm quite sure you ask yourself this question. Please note that all templates and macros that use icon filter (as before with FontAwesome i-tag-classes) are still wrapped (in most cases) in Bulma-specific CSS container elements (mostly span-tags). This will change in another PR derived from #1035. 

I hope this clarifies your question. If not, feel free to ask further questions. I'm always happy to clear things up ☺️

Reply:
> Thanks for the explanation. I did not mean to imply that the design is bad. I first of all wanted to understand if there is a valid "use case" for passing an empty string into this filter because I am not very familiar with the current setup. I makes sense to be compatible with "historic" design decisions.

You are welcome. I'm glad, I could clarify the "historic" design decision context ☺️ Did you already spot the concerning documentation for icon filter [in our docs](https://arbeitszeitapp.readthedocs.io/en/latest/development_guide.html#icon-templates-integration-and-usage)? If not, maybe worth checking out, to better understand the icon template creation und icon filter usage. If you find stuff to improve in there, please let me know. I'm happy to update the concerning section(s)!